core/zip/inc/Compression.h

```diff
@@ -43,91 +43,94 @@ namespace ROOT {
-         kDefaultOld = 6,
-         /// Compression level reserved for LZMA compression algorithm (slowest compression with smallest files)
-         kDefaultLZMA = 7
-      };
+
+   enum class ELevel {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object
+      kInherit = -1,
+      /// Compression level reserved for "uncompressed state"
+      kUncompressed = 0,
+      /// Compression level reserved when we are not sure what to use (1 is for the fastest compression)
+      kUseMin = 1,
+      /// Compression level reserved for ZLIB compression algorithm (fastest compression)
+      kDefaultZLIB = 1,
+      /// Compression level reserved for LZ4 compression algorithm (trade-off between file ratio/decompression speed)
+      kDefaultLZ4 = 4,
+      /// Compression level reserved for ZSTD compression algorithm (trade-off between file ratio/decompression
+      /// speed)
+      kDefaultZSTD = 5,
+      /// Compression level reserved for old ROOT compression algorithm
+      kDefaultOld = 6,
+      /// Compression level reserved for LZMA compression algorithm (slowest compression with smallest files)
+      kDefaultLZMA = 7
    };
-   struct EAlgorithm { /// Note: this is only temporarily a struct and will become a enum class hence the name
-                        /// convention used.
-      enum EValues {
-         /// Some objects use this value to denote that the compression algorithm
-         /// should be inherited from the parent object (e.g., TBranch should get the algorithm from the TTree)
-         kInherit = -1,
-         /// Use the global compression algorithm
-         kUseGlobal = 0,
-         /// Use ZLIB compression
-         kZLIB,
-         /// Use LZMA compression
-         kLZMA,
-         /// Use the old compression algorithm
-         kOldCompressionAlgo,
-         /// Use LZ4 compression
-         kLZ4,
-         /// Use ZSTD compression
-         kZSTD,
-         /// Undefined compression algorithm (must be kept the last of the list in case a new algorithm is added).
-         kUndefined
-      };
+
+   enum class EAlgorithm {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object (e.g., TBranch should get the algorithm from the TTree)
+      kInherit = -1,
+      /// Use the global compression algorithm
+      kUseGlobal = 0,
+      /// Use ZLIB compression
+      kZLIB,
+      /// Use LZMA compression
+      kLZMA,
+      /// Use the old compression algorithm
+      kOldCompressionAlgo,
+      /// Use LZ4 compression
+      kLZ4,
+      /// Use ZSTD compression
+      kZSTD,
+      /// Undefined compression algorithm (must be kept the last of the list in case a new algorithm is added).
+      kUndefined
    };
 
-   static std::string AlgorithmToString(EAlgorithm::EValues algorithm);
+   static const char *AlgorithmToString(EAlgorithm algorithm);
```

Comment:
This is a user code breaking change.  Is it more than an optimization (of a rarely run function)?

Reply:
Is it really a breaking change? `std::string` can be implicitly constructed from a `const char *`, so in which case does this make a difference user-side?

Reply:
The legal
```
something_needing_const_char_star(  AlgorithmToString(value).c_str() );
```
will fail.